restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisIncomingPhoneNumbersDao.java

```diff
@@ -70,19 +74,61 @@ public IncomingPhoneNumber getIncomingPhoneNumber(final String phoneNumber) {
         return getIncomingPhoneNumber("getIncomingPhoneNumberByValue", phoneNumber);
     }
 
-    private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
+   private IncomingPhoneNumber getIncomingPhoneNumber(final String selector, Object parameter) {
         final SqlSession session = sessions.openSession();
-        try {
+        String inboundPhoneNumber = null;
+          try {
+            //if(parameter.toString().contains("*")){}{
+            //perform direct check
             final Map<String, Object> result = session.selectOne(namespace + selector, parameter);
-            if (result != null) {
+            if (result != null ) {
                 return toIncomingPhoneNumber(result);
-            } else {
-                return null;
-            }
-        } finally {
+            }//}
+            //check if there is a Regex match only if parameter is a String aka phone Number
+            if(!(parameter instanceof Sid)){
```

Comment:
@croufay why you check if object parameter is not an instance of Sid? What is the point here?